### PR TITLE
Fix #7204: add a dedicated source parameter to JSON objects

### LINKED ISSUE #7204: RCT1 object source filters from #7106 are not functional
**OS:** Windows 10 (x86-64)
**Version:** 0.1.2
**Commit/Build:** d5d9441

#7106 
capture

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/object/Object.h [hunk 1]
```diff
@@ -148,6 +148,8 @@ class Object
     rct_object_entry    _objectEntry{};
     StringTable         _stringTable;
     ImageTable          _imageTable;
+    uint8_t             _sourceGame = OBJECT_SOURCE_CUSTOM;
+    uint8_t             _secondSourceGame = OBJECT_SOURCE_CUSTOM;
```
Comment (IntelOrca) [inline src/openrct2/object/Object.h]: I'd like to see a `std::vector` used instead.

File: src/openrct2/object/Object.h [hunk 2]
```diff
@@ -183,10 +182,8 @@ class Object
     virtual std::string     GetName(int32_t language) const;
 
     virtual void  SetRepositoryItem(ObjectRepositoryItem * /*item*/) const { }
-    uint8_t       GetSourceGame();
-    void          SetSourceGame(const uint8_t sourceGame);
-    uint8_t       GetSecondSourceGame();
-    void          SetSecondSourceGame(uint8_t sourceGame);
+    std::vector<uint8_t> GetSourceGames();
+    void          SetSourceGames(std::vector<uint8_t> sourceGames);
```
Comment (IntelOrca) [inline src/openrct2/object/Object.h]: `const std::vector<uint8_t>&`

**Comment by IntelOrca** [comment]
You call `SetPrimarySceneryGroup` for footpath items and banners. Do you still need to do it for WW and TT sources? If you always want it to appear in the group then you don't need to do the check, you can just enforce it.

**Comment by Gymnasiast** [comment]
The check is there to avoid items appearing twice. Basically, if SCGPATHX includes an item, and the item requests being put in SCGPATHX as well, it will appear twice.

I can change the WW and TT and reduce the check to just "custom objects".

**Comment by IntelOrca** [comment]
Doesn't that mean that if a custom object lists the footpath category it would appear twice? We should probably just improve the code to only add items once to a category.

**Comment by Gymnasiast** [comment]
> Doesn't that mean that if a custom object lists the footpath category it would appear twice?

No. There are two ways of making an object appear in a scenery group:
1. Have the scenery group list the object (the most common method).
2. Have the object list the scenery group.

As you can see, it's possible to create circular references this way. If you do so, an object will be listed twice.

**Comment by IntelOrca** [comment]
Okay, could you for now just change it to the following then?
```cpp
if (item->Sources.empty() || items->Sources[0] == OBJECT_SOURCE_CUSTOM)
{
    ...
}

**Pull request merged**